bids/grabbids/bids_layout.py

```diff
@@ -40,6 +40,17 @@ def __init__(self, path, config=None, validate=False,
         super(BIDSLayout, self).__init__(path, config=config,
                                          dynamic_getters=True, **kwargs)
 
+    def __repr__(self):
+        n_sessions = len([session for isub in self.get_subjects()
+                          for session in self.get_sessions(subject=isub)])
+        n_runs = len([run for isub in self.get_subjects()
```

Comment:
This is fine, but if I'm not mistaken, you may be able to replace the loops with a single grabbit query like `n_runs = len(self.get(target='run', return_type='dir'))`. Querying grabbit is currently quite inefficient (every query searches exhaustively over all files), so this might speed things up very slightly (though it would probably only be noticeable for large projects).

Reply:
hmm, I just tried running `layout.get(target='run', return_type='dir')` but got the error

`ValueError: Return type set to directory, but no directory template is defined for the target entity ("run").`

It's a bit opaque to me how the functionality changes when you go from `return_type='tuple'` to `return_type='dir'`...it seems like it requires a different kind of input in this case? so maybe I'm messing this up...

Reply:
(either way, I rebased and just pushed so maybe that'll make tests happy)

Reply:
I was about to do something similar, glad I ran into this.

I'd also add `Name` from dataset_description.json
```python
        descr_file = self.get(type='description')
        with open(descr_file[0].filename) as dsf:
            descr = json.load(dsf)
        s = "BIDS Layout: ...{}\nName: {} | Subjects: {} | Sessions: {} | Runs: {}".format(
                root, descr['Name'], n_subjects, n_sessions, n_runs)

```

Reply:
Looks like you're right, my suggestion won't work. I think we can still improve the efficiency of this, but it's fine for now. Okay to merge, @choldgraf?

Reply:
I think we're ready to merge as long as the suggestion from @raamana isn't super simple to implement. They mentioned including the "name" of the dataset in the repr, but from what I can tell this isn't straightforward. I dug around the API to try to find a `description` attribute or something like this, but it doesn't seem like pybids checks for this or loads it automatically. It seems like extending the repr method to include the "name" should happen if/when pybids requires and checks that the file path a Layout gets also adheres to the BIDS spec (aka, that it has a `__description.json` file in it) does that make sense?

Reply:
I agree, we shouldn’t complicate it. I was just coming from the succinct and yet complete info about the dataset.

Reply:
@tyarkoni I think we're +1 on a merge in this case!

@raamana I definitely agree something like "Name: " should be incorporated once that piece of metadata is quickly/reliably exposed with a Layout instance

Reply:
Yes, adding name would be nice. But this should actually be a separate (and frankly, higher priority) issue. I didn't actually realize that the `Name` and `BIDSVersion` fields were mandatory; pybids should definitely read those in!